src/cascadia/TerminalSettingsEditor/MainPage.cpp

```diff
@@ -645,4 +647,58 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
         return SettingsNav().Background();
     }
 
+    bool MainPage::OnDirectKeyEvent(const uint32_t vkey, const uint8_t /*scanCode*/, const bool down)
+    {
+        auto handled = false;
+        const auto modifiers = _GetPressedModifierKeys();
+        if (modifiers.IsAltPressed() && vkey == VK_SPACE && down)
+        {
+            _OpenSystemMenuHandlers(*this, nullptr);

```

Comment:
We _should_ go through the keybindings here but SUI doesn't have access to them. We have this bound in TerminalPage instead to enable key bindings.

Fun fact: the command palette has the same exact problem

I was working on this but it ended up being too risky so I decided to just send this PR here. But I could...
- introduce a `struct DirectKeyEventArgs`
- in CommandPalette and SUI (MainPage), raise a custom `DirectKeyPressed` event that hands the populated `DirectKeyEventArgs` over to TerminalPage
- refactor `TerminalPage::KeyDownHandler` (or whatever it's called) to be called when this event is dispatched

It's riskier because then we hit some bugs (particularly in the command palette) when the system menu is opened then closed (key events seem to get lost after that?). But then we _definitely_ have alt+space run the bound command rather than just a hardcoded "open system menu".

@DHowett thoughts?

Reply:
> SUI doesn't have access to them.

UGH SERIOUSLY?